lms/djangoapps/mobile_api/course_info/views.py

```diff
@@ -153,6 +154,12 @@ def post(self, request, *args, **kwargs):
                 status=status.HTTP_400_BAD_REQUEST,
             )
 
+        if not COURSE_GOALS_NUMBER_OF_DAYS_GOALS.is_enabled(course_key):
+            return Response(
+                'User activity is not enabled for this user and course',
+                status=status.HTTP_400_BAD_REQUEST,
```

Comment:
this doesn't seem like the right response code.  This would occur when the mobile apps try to tell the backend about the user having done some activity but for some reason the flag is not enabled for the request, correct?  And those reasons could be a user- or course-specific exemption, or that we've shut off the feature completely for some reason.  Seems to me that the mobile app shouldn't be bothered with 400 response codes in that case.  

Maybe log a warning that these POSTs are being ignored, but then return a 200 response code to the caller?

Reply:
hm, it feels like the log warnings are likely to just go off into a void, but on the other hand, it shouldn't be a surprise this wouldn't work if the entire feature is disabled, so i think returning a 200 with a log sounds good to me - updated